### DISCUSSION TITLE
Is it possible to capture all the collision and fragmentation processes as XYZ files?

### DISCUSSION THREAD

**Discussion by tobigithub** [qcxms/QCxMS#28] [Ideas]
Hi,
I was running a CID simulation of [scopolamine](https://pubchem.ncbi.nlm.nih.gov/compound/Scopolamine), but not all the collision steps are actually captured. In the two major steps no fragmentation occurs, captured as trajectory files for Avogadro or Jmol (CID1.xzy and CID2.xyz). In the third CID3.xyz file the fragmentation starts immediately with a smaller fragment, but one can not actually visualize the collision, its only captured in the MD files which are (hidden) xyz files, without the collision gas. So if the dissociation occurs after CID2, then it would be nice to automatically rename the "MDtrj.1.2.2" file also into XYZ. Also the specific filenames could directly added to the qcxms.out file to easier find associated reactions. The naming schema gives it away, but its not always easy to decipher the order of those files. 

From a mass spectrometric perspective I think the visualization of all CID fragmentation processes is very important, especially those where the molecules collide, where the bond breaks, which additional rearrangements occur. So I was wondering if this could be enabled as a future feature, I added it here as an idea in discussions, in order to avoid spamming the issue board. I added the visual example and the trajectory files here [scopolamine-minibench.tar.gz](https://github.com/qcxms/QCxMS/files/7087052/scopolamine-minibench.tar.gz).

Tobias

![image](https://user-images.githubusercontent.com/2569565/131582300-cb7c915c-b90f-4e67-a8b7-ba2f2c5f592b.png)

````
  
================================================================================
      trajectory     1      collision   2 /14
================================================================================

...snip...snip
 
 E X I T - the CID module is finished
 
 fragment assigment list:1111111111111111111111111111111111111111111
 computing average fragment structures ...
  mass                formula                  q pop   spin    |q IPB|  diss time (ps)
 M=303.35           H21C17N1O4     1   2   1   0.961   0.000   1.000    0.000
 

================================================================================
   Collision            2  MD trajectory                       2
================================================================================

 initial Cartesian coordinates:
  1   6.92125   1.16544  -4.23243    N     14.007
  2   4.26007   1.24623  -4.36437    C     12.011
  3   6.80209   3.21949  -1.87281    C     12.011
  4   5.60709   1.82377   0.27532    C     12.011
  5   2.58822  -0.72239  -2.92641    C     12.011
  6   3.84650  -0.62525  -0.22077    C     12.011
  7   8.50781   1.02322  -6.25452    C     12.011
  8   1.59298  -0.15102   1.94855    O     15.999
  9  -0.37141  -1.75951   1.41019    C     12.011
 10  -0.26400  -3.39452  -0.08731    O     15.999
 11  -2.66428  -0.46173   3.28826    C     12.011
 12  -1.46230  -1.93290   5.41123    C     12.011
 13  -5.70708  -1.08286   2.39340    C     12.011
 14  -1.10173  -4.35290   4.65201    O     15.999
 15  -6.40879  -1.36596   0.05893    C     12.011
 16  -9.17095  -0.89047  -0.36125    C     12.011
 17 -10.91663  -1.24092   1.52696    C     12.011
 18  -9.81910  -2.36378   4.01264    C     12.011
 19  -7.33973  -1.88646   4.51362    C     12.011
 20   4.99576   5.22145  -2.60821    C     12.011
 21   3.09643   4.11248  -4.16486    C     12.011
 22   2.55372   4.75034  -1.61058    O     15.999
 23   3.64619   0.61861  -6.43022    H      1.008
 24   8.60891   3.71462  -1.44833    H      1.008
 25   4.30921   2.98233   1.47480    H      1.008
 26   6.87700   0.58038   1.49454    H      1.008
 27   2.39001  -2.95138  -3.19826    H      1.008
 28   0.88280  -0.32903  -2.27471    H      1.008
 29   4.65830  -2.55943   0.22465    H      1.008
 30  10.34772   0.93274  -5.68484    H      1.008
 31   7.93646   2.72953  -7.71963    H      1.008
 32   8.60700  -0.55535  -7.49981    H      1.008
 33  -2.03928   1.55674   3.49340    H      1.008
 34   0.34825  -0.74023   5.80858    H      1.008
 35  -2.47159  -2.33274   7.67280    H      1.008
 36  -2.03444  -5.11659   3.24576    H      1.008
 37  -5.11669  -0.62341  -1.52928    H      1.008
 38 -10.12729  -0.54537  -2.34601    H      1.008
 39 -12.90904  -0.34552   1.34610    H      1.008
 40 -11.18184  -3.41208   4.90000    H      1.008
 41  -7.10286  -2.54985   6.57762    H      1.008
 42   6.06475   6.90798  -2.71846    H      1.008
 43   2.54784   5.17825  -5.68718    H      1.008

 statistical charge  =     1.0000

 
 initializing GFN2-xTB ...
 initialization successful!
 energy =      -65.49863
 charge =              1
 mult   =              2
 etemp  =         5000.0

             E N T E R I N G   M D   M O D U L E

Eimp (eV) =    0.0     tauIC (fs) =     0.      nstep =    8000

avcycle =   50    more =  250

step   time [fs]    Epot       Ekin       Etot    error  #F   eTemp   frag. T
      0   6685.    -65.49863   0.3629   -65.13571  0.0000 1    5000.    1777.
    100   6735.    -65.57438   0.4295   -65.14484  0.0040 1    5000.    1945.
    200   6785.    -65.48494   0.3361   -65.14886  0.0089 1    5000.    1916.
    300   6835.    -65.44704   0.3240   -65.12305 -0.0169 1    5000.    1868.
    400   6885.    -65.50546   0.3640   -65.14149  0.0014 1    5000.    1838.
    500   6935.    -65.54316   0.4003   -65.14291  0.0027 1    5000.    1842.
    600   6985.    -65.51782   0.3701   -65.14774  0.0076 1    5000.    1844.
    700   7035.    -65.51181   0.3713   -65.14053  0.0004 1    5000.    1839.
    800   7085.    -65.45818   0.3231   -65.13511 -0.0050 1    5000.    1846.
    900   7135.    -65.52689   0.3945   -65.13243 -0.0077 1    5000.    1840.
   1000   7185.    -65.45201   0.3093   -65.14273  0.0027 1    5000.    1832.
   1100   7235.    -65.49386   0.3512   -65.14261  0.0025 1    5000.    1826.
   1200   7285.    -65.50669   0.3755   -65.13124 -0.0089 1    5000.    1830.
   1300   7335.    -65.55659   0.4200   -65.13657 -0.0036 1    5000.    1836.
   1400   7385.    -65.52932   0.3886   -65.14073  0.0006 1    5000.    1831.
   1500   7435.    -65.53041   0.3966   -65.13383 -0.0063 1    5000.    1828.
   1600   7485.    -65.54005   0.4047   -65.13539 -0.0047 1    5000.    1825.
   1700   7535.    -65.53268   0.3797   -65.15302  0.0129 1    5000.    1821.
   1800   7585.    -65.47124   0.3433   -65.12798 -0.0121 1    5000.    1822.
   1900   7635.    -65.55676   0.4115   -65.14526  0.0051 1    5000.    1823.
   2000   7685.    -65.45916   0.3215   -65.13769 -0.0024 1    5000.    1824.
   2100   7735.    -65.53179   0.4104   -65.12138 -0.0187 1    5000.    1826.
   2200   7785.    -65.46053   0.3235   -65.13708 -0.0030 1    5000.    1827.
   2300   7835.    -65.54304   0.3987   -65.14431  0.0042 1    5000.    1831.
   2400   7885.    -65.54865   0.4066   -65.14207  0.0020 1    5000.    1835.
   2500   7935.    -65.48312   0.3587   -65.12444 -0.0156 1    5000.    1839.
   2600   7985.    -65.49811   0.3541   -65.14398  0.0039 3    5000.   11046. 21112.  9169.
   2700   8035.    -65.52430   0.3823   -65.14201  0.0019 3    5000.   11661. 18415.  9185.
   2800   8085.    -65.51703   0.3721   -65.14498  0.0049 3    5000.   11966. 18286.  8721.
   2813   8092.    -65.56694   0.4230   -65.14398  0.0085 3    5000.   12283. 18773.  8860.

        E X I T   M D  because of nfrag=nfragexit

          Results
 average Ekin     0.374759
 average Epot    -65.514851
 average Etot    -65.140093
 average T (acc.)      1834.7
 Temp without acc       1904.0
 average last T      3916.9
 fragment assigment list:1111111222333333333111111111111133333333311
 computing average fragment structures ...
  inter fragment distances (Angst.)

           1         2         3
 
    1   0.00000
    2   6.39739   0.00000
    3   8.88952   5.62052   0.00000

 computing IPs with       GFN2-xTB at (K)   3917.
 fragment  1 E(N)=    -30.0387  E(I)=    -29.6456             IP/EA(eV)=   10.70
 fragment  2 E(N)=    -10.3006  E(I)=     -9.5840             IP/EA(eV)=   19.50
 fragment  3 E(N)=    -25.5482  E(I)=    -25.1140             IP/EA(eV)=   11.81

 wall time for IP (s)       0.0

  mass                formula                  q pop   spin    |q IPB|  diss time (ps)
 M=138.19            H12C8N1O1     1   2   2   0.625   0.000   0.965    7.948 ~
 M=44.01                  C1O2     1   2   2  -0.012   0.000   0.000    7.948
 M=121.16               H9C8O1     1   2   2   0.387   0.000   0.035    7.948
 
 
 Number of frag Processes           1
 ================================================================================
````

**Comment by JayTheDog** [comment]
Hi Tobias,

I'm not quite sure what you are asking for. As you can see in your example, the fragmentation occurs in the mean-free-path MD, after the second collision and before the third collision. In this case, the energy transfer from collision-to-internal energy takes some time to be distributed into the vibrational frequencies, so that the second collision is enough to induce the fragmentation, but just "needs" more time. This mean-free-path (MFP) is obviously shorter than in the experiment, but can be a huge factor in computational time. In these MDs, no collision gas atom is sampled, because 1) the ion-gas distance is very large, i.e. no influence of the gas atom, 2) the xyz files would explode in size and 3) would also affect the visualization of the MD. 

I'm not quite sure what you mean with "hidden" files. The MFP MD, which is written in the MDtrj.1.2.2 file, is in the same format as the EI MD trajectory, so these are .xyz files and should easily be opened with all visualization programs. I agree that naming the CID steps **.xyz** while not doing the same for the MDs is not very consequential, but was planned to better discriminate between xyz files of the fragments (e.g. 3.1.2.xyz) and the MDs (either CID MD or MFP MD). From my logic, it would make more sense to rename the *CID.xyz* files to only *CID* to better discriminate. What do you think?

All fragments and ions (even if not fragmented) are always written out after each collision, as well as each MFP MD. 
The naming of the .xyz files can take some getting used to, but follows the logic [collision].[run-number].[fragment-number].xyz . 

So all processes are captured and you can see where exactly fragmentation occurs. In your case, the ion gains enough energy to dissociate in the second step, but not in a **direct** fragmentation (the exact moment of the collision leads to fragmentation), but rather in a **statistical** fragmentation (internal energy conversion). These statistical fragmentation are better in capturing rearrangement reactions, so having a collision as well as a MFP MD is only beneficial for the simulation.

**Comment by tobigithub** [comment]
@JayTheDog  Thanks. 

So I would not rename the CID files, also the numbering is OK, its just hard to understand and decipher for novice users that have never worked with QCxMS. All your explanations sound logical to me.

Basically I am concerned with (1) ease of use for new users and (2) provenance that can be easily followed.
So besides creating MS/MS spectra, the investigation of CID processes is important. So when investigating CID processes, the user should be able to follow them easily, without deciphering file names. That can be solved in 2-3 ways:

1) add the XYZ extension to the MD files, so its visible for (inexperienced) users they can be animated as xyz files (ease-of-use).

2) add the CID and MD filenames that were created to the qcxms.out file, exactly when the individual runs are finished and logged to qcxms.out.  In this way there is a  connection to the log parameters and the filenames that were created (provenance and ease-of-use). 

3) Add a simple table with all the CID and MD processes, for example as below (times are not necessary, just the order). For the CID processes its clear [1..9], but the numbering of MD file names just becomes harder to read when going into higher trajectory counts. Its a tiny simple thing, but this could be useful for ease-of-use and also provenance. 

| #  | time  | File created |
|----|-------|--------------|
|  1 | 13:35 |  MDtrj.1.0.1 |
|  2 | 13:36 |   CID1.xyz   |
|  3 | 13:41 |  MDtrj.1.1.2 |
|  4 | 13:42 |   CID2.xyz   |
|  5 | 13:44 |  MDtrj.1.2.2 |
|  6 | 13:45 |  MDtrj.1.2.3 |
|  7 | 13:45 |   CID3.xyz   |
|  8 | 13:46 |  MDtrj.1.3.2 |
|  9 | 13:47 |   CID4.xyz   |
| 10 | 13:48 |  MDtrj.1.4.2 |
| 11 | 13:48 |   CID5.xyz   |
| 12 | 13:50 |  MDtrj.1.5.2 |
| 13 | 13:50 |   CID6.xyz   |
| 14 | 13:51 |  MDtrj.1.6.2 |
| 15 | 13:52 |   CID7.xyz   |
| 16 | 13:53 |  MDtrj.1.7.2 |
| 17 | 13:53 |   CID8.xyz   |
| 18 | 13:54 |  MDtrj.1.8.2 |
| 19 | 13:54 |   CID9.xyz   |
| 20 | 13:55 |  MDtrj.1.9.2 |

**Reply by awvwgk** [reply]
I'm confused, why do you ping me in this thread?

**Reply by tobigithub** [reply]
Sorry that was for @JayTheDog
